### ISSUE TITLE
Amazon … video can not be viewed in HD (1080p) – content restrictions – in some environments

### ISSUE THREAD

**Issue by grahamperrin** [MrAlex94/Waterfox#1216]
### 2019-07-05: _Amazon Prime_ (<strike>region unknown</strike> US) not working in HD

[Your browser does not support High-Definition playback due to content restrictions, please update your browser to get the best Prime Video experience. : waterfox](https://www.reddit.com/r/waterfox/comments/c9oe2e/your_browser_does_not_support_highdefinition/)

https://www.reddit.com/comments/c9oe2e/-/eti4bne/?context=2 in particular: 

> All I know is, it was working on July 4th. On July 5th, I had this message come up. Nothing changed. I didn't have any updates from Windows Update, I didn't install any new programs or extensions or anything. Nothing changed. Absolutely nothing - except for this message on Amazon Prime Videos. I only installed 56.2.11 in **response** to this problem hoping that it would solve it.

OP: [/u/TwoCables_from_OCN](https://www.reddit.com/user/TwoCables_from_OCN)

I assume that Waterfox Classic 56.2.10.1 (2019-06-07) was in use when the restriction first appeared. 

### 2019-07-15: _Amazon Prime India_ working in HD

For a problem with HD that was [reported on 2019-06-29](https://www.reddit.com/r/waterfox/comments/c5khws/amazon_prime_video_india_not_working/esb6qj2/), around [two weeks later](https://www.reddit.com/r/waterfox/comments/c5khws/amazon_prime_video_india_not_working/ettw77h/):

> … working , even HD …
> 
> Used this string
> 
> `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:67.0) Gecko/20100101 Firefox/67.0`

### 2019-08-30: _Amazon Prime_ (region unknown) not working, then working, in HD

[Anyone else use Youtube TV and Amazon Prime having issues? : waterfox](https://www.reddit.com/r/waterfox/comments/cxfpg5/anyone_else_use_youtube_tv_and_amazon_prime/)

From the opening post: 

>>> 56.2.13 … Amazon Prime … doesn't support HD 

– and later the same day from the OP: 

>> got Prime working with User Agent 

`/u/TwoCables_from_OCN` observed: 

> … HD … User Agent didn't fix that for me.

### Other discussions 

https://www.reddit.com/comments/ctg797/-/exm69yn/?context=1 (2019-08-21, `/u/TwoCables_from_OCN`) with uncertainty https://www.reddit.com/comments/ctg797/-/exrn0fj/ about the relationship to the opening post. 

[56.2.14 still doesn't play Amazon Prime Videos in HD : waterfox](https://www.reddit.com/r/waterfox/comments/d20dos/56214_still_doesnt_play_amazon_prime_videos_in_hd/) (2019-09-10, `/u/TwoCables_from_OCN`) 

https://www.reddit.com/comments/deuzev/-/f31pz23/ (2019-10-09, `/u/TwoCables_from_OCN`) under a post about Waterfox Current 68.0b1. 

Any others?

### Unknowns

It might help to know the addresses of some affected videos<strike>, and a region</strike>. 

### Meta, tracking

#538

**Comment by grahamperrin** [comment]
Can anyone reproduce this issue with Waterfox Current?

**Comment by grahamperrin** [comment]
>> It might help to know the addresses of some affected videos…

https://www.reddit.com/comments/c9oe2e/-/f4ekvw7/?context=2

> Any video on Amazon Prime Video that is in HD. Any. Pick one and go for it.
> 
> I live in the U.S.

I don't have Amazon Prime, I do have just one Amazon Video https://www.amazon.co.uk/gp/video/detail/B00X674KKU 

![image](https://user-images.githubusercontent.com/192271/67159362-5408e580-f33b-11e9-9cba-78626d786d18.png)

- available in HD
- I paid £4.99 (UK) so I _assume_ that whenever I found it playing (in Waterfox Classic on Windows 10), playback was HD … honestly, I never paid attention to the quality in the virtual machine. 

Retesting now with Waterfox Classic 2019.10 (20191016181224), success: 

https://put.re/player/CS4NUPKJ.34.mp4

A frame from the recording: 

![Amazon Video, best quality, around 1 17 GB per hour](https://user-images.githubusercontent.com/192271/67159535-d1cdf080-f33d-11e9-8885-4c27c8f5a98c.png)

- best quality – <strike>is this HD? </strike>HD, not _HD 1080p_.

The same video in Firefox ESR 60.9.0 in the same virtual machine: 

![image](https://user-images.githubusercontent.com/192271/67159836-e2806580-f341-11e9-970f-ead11385e1d5.png)

**Comment by grahamperrin** [comment]
From <https://www.reddit.com/r/waterfox/comments/deuzev/-/f4ir5pg/>: 

> … it seems I’ve managed to get some videos to play in HD again - seeing what I can do to get it to work on Classic branch.

– from which, I assume that issues affecting HD or HD (1080p) are not limited to Waterfox Classic.

**Comment by heliosfa** [comment]
I have just run across this issue in Waterfox Classic 2020.01 (64-bit) on Windows 10 1909. Honestly, it is the first time I have tried to watch Prime Video content in Waterfox for a while and I get the "Your browser does not support High-Definition playback due to content restrictions" message when trying to play [Episode 1 of Star Trek Picard on Amazon UK](https://www.amazon.co.uk/gp/product/B083C3Z4NW?autoplay=1).
![wf56](https://user-images.githubusercontent.com/6616536/73110053-eb627380-3efc-11ea-93da-04899d883471.PNG)


 

The desired behaviour would be HD playback as can be achieved in Chrome and Firefox. Chrome 79.0.3945.130 will display in `HD 1080p` in the bottom left of the screen during full screen playback and `HD` (e.g. 720p) when windowed.
![Chrome1080p](https://user-images.githubusercontent.com/6616536/73109620-907c4c80-3efb-11ea-9378-b8e7fefff380.PNG)
![ChromeHD](https://user-images.githubusercontent.com/6616536/73109625-92dea680-3efb-11ea-8e21-7692c1e4a407.PNG)

Firefox 72 starts off in `HD` in both windowed and fill-screen, and will sometimes end up in `HD 1080p` after a few (tens of) seconds playback.

When playing back in HD, the "best" video quality should use "about 6.84 GB per hour"
![playback](https://user-images.githubusercontent.com/6616536/73110000-c1a94c80-3efc-11ea-96ab-8ef07bc5e367.png)

&nbsp;


Removing `Waterfox/56.3` from the UserAgent (e.g. changing it to `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0 Waterfox/56.3`) completely blocks video playback with a "Mozilla Firefox Update Required" screen:
![ff56](https://user-images.githubusercontent.com/6616536/73110093-09c86f00-3efd-11ea-95cd-b491026fb4c2.PNG)

&nbsp;


I decided to do a bit of futzing with the useragent to see if I could find something that worked. Unfortunately, everything that I have tried did not remove the message or broke playback completely. I tried::
- Firefox 67 as suggested further up `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:67.0) Gecko/20100101 Firefox/67.0`
- latest Firefox `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:72.0) Gecko/20100101 Firefox/72.0`
- latest Chrome `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36`
- Latest Opera `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.123 Safari/537.36 OPR/66.0.3515.44`
- Latest Edge `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.102 Safari/537.36 Edge/18.18363` (this one threw the "Unsupported Browser" error)
![EdgeFail](https://user-images.githubusercontent.com/6616536/73110114-1ea50280-3efd-11ea-89a3-fc026b793bd7.PNG)
- Latest Safari `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Safari/605.1.15` resulted in a black screen with no playback
- Firefox Mac `Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:72.0) Gecko/20100101 Firefox/72.0`

&nbsp;


In other words, it looks like a user-agent only fix like exists for Netflix on Mac is not possible here.

Hopefully this gives some more insight into this issue - if you need any more info or more testing, I am happy to help.

**Comment by heliosfa** [comment]
Further to my last comment on this, I can confirm that the same issue is present on Mac.

**Comment by RocketBun** [comment]
just here to confirm this issue is still present on latest windows release.

**Comment by Roykirk1701** [comment]
> 
> 
> Can anyone reproduce this issue with Waterfox Current?

Yes, it occurred for me today with Current 2020.02

**Comment by hawkeye116477** [comment]
@grahamperrin HD is not available generally for Linux and Waterfox Classic (on all OS). Best quality isn't always HD, when is HD then on bottom left you should see `HD` mark. Now you can test on free Prime videos.

**Comment by oblivion2k** [comment]
No comments in several months, chiming in to report that 2020.07.2.1 (64-bit) current branch, latest update still exhibits this issue.

**Comment by mikeloeven** [comment]
At this point I have to say the issue is likely malicious action on Amazons part they are clearly doing something more invasive than simply checking the user agent. As Waterfox does have the same capabilities of the current Firefox at this point I feel we need to make an effort to figure out what amazon is checking and implement stealth countermeasures in the code to ensure Amazon will not be able to detect that Waterfox is not a standard FF build.

 As stated in https://github.com/MrAlex94/Waterfox/issues/985

there should already be a UA override in place for amazon so it is doing something more invasive than simply checking UA

**Comment by yarow12** [comment]
I've been having this problem for almost/atleast a year. USA, Windows 7 64bit, Current G3.0.2, Ethernet connected, and the only thing that _should_ be connected to my wifi is my phone. 
Was using Classic until Reddit's update became incompatible with it last year. No recollection of whether or not it had that problem then, but I just checked Classic v2020.09 and it says "HD Unavailable".

Back to Current: YouTube's 1080 HD seems to work just fine. MUBI, on the other hand, switches back and forth between HD and SD. Chrome (v88.0.4324.104) and my phone don't have this problem. 
You can test this with Amazon Video's trailers, btw.

If this message is incoherent, it's because I'm tired.

**Comment by ombrisban** [comment]
I'm using Waterfox G3.0.2 (64-bit)  in Windows 10 and I have de same issue...

![screenshot-86](https://user-images.githubusercontent.com/64747841/107095716-93e22d80-6809-11eb-9c6b-eb09d891404d.jpg)

Amazon Prime does not present the option to configure the best quality (more than 6GB / hour) 

![screenshot-87](https://user-images.githubusercontent.com/64747841/107096056-28e52680-680a-11eb-89c5-9e34edb717b9.jpg)

**Comment by mikeloeven** [comment]
@MrAlex94 Has there been any progress made regarding this issue. The ticket has been open for over a year at this point. Being such a major wildly used service this is becoming an increasingly wide spread problem with steaming providers

**Comment by mikeloeven** [comment]
@MrAlex94 Can this ticket be reopened I don't see any comments or reason entered for why it was closed and i can confirm the issue persists in the most recent version of the browser
